tools/rc/ResourceCompiler.cpp

```diff
@@ -110,6 +112,178 @@ std::string us_tempfile()
 
 #endif
 
+// ---------------------------------------------------------------------------------
+// ------------------------   END PLATFORM SPECIFIC CODE    ------------------------
+// ---------------------------------------------------------------------------------
+
+namespace 
+{
+
+class InvalidManifest : public std::runtime_error
+{
+public:
+  InvalidManifest(const std::string& msg) : std::runtime_error(msg) {}
+  InvalidManifest(const char* msg) : std::runtime_error(msg) {}
+};
+
+/*
+ * @brief parses json content and returns the parsed json or throws.
+ * @tparam jsonContent json content to parse. The type must be convertible to std::istream.
+ * @param root The parsed Json root object.
+ * @throw InvalidManifest if the json is invalid. Parse error information is in the exception.
+ * If an exception is thrown, the root param is invalid.
+ */
+template <class T>
+void parseAndValidateJson(T& jsonContent, Json::Value& root)
+{
+  Json::CharReaderBuilder rbuilder;
+  rbuilder["rejectDupKeys"] = true;
+  rbuilder["allowComments"] = false;
+  std::string errs;
+
+  if (!Json::parseFromStream(rbuilder, jsonContent, &root, &errs))
+  {
+    throw InvalidManifest(errs);
+  }
+}
+
+/*
+ * @brief parses json content from a file and returns the parsed json or throws.
+ * @param jsonFile path to a json file.
+ * @param root The parsed Json root object.
+ * @throw InvalidManifest if the json is invalid. Parse error information is in the exception.
+ * If an exception is thrown, the root param is invalid.
+ */
+void parseAndValidateJsonFromFile(const std::string& jsonFile, Json::Value& root)
+{
+  try
+  {
+    std::ifstream json(jsonFile);
+    parseAndValidateJson(json, root);
+  }
+  catch (const InvalidManifest& e)
+  {
+    std::string exceptionMsg(jsonFile + ": " + e.what());
+    throw InvalidManifest(exceptionMsg);
+  }
+}
+
+/* 
+ * @brief Validate manifest files in an archive.
+ * @param archiveFile archive file path
+ * @throw std::InvalidManifest on the first invalid manifest found.
+ */
+void validateManifestsInArchive(const std::string archiveFile)
```

Comment:
Probably doesn't matter at all but to be consistent with your other signatures

`const std::string archiveFile` -> `const std::string& archiveFile`